### ISSUE TITLE
Accuracy metric behaviour change between versions 0.9.2 and 0.10.0 (and later)

### ISSUE THREAD

**Issue by Kr4is** [Lightning-AI/metrics#1444]
## 🐛 Bug

Using pytorch lightning, obtaining the accuracy metric in the "test_step" with torchmetrics version 0.10.0 or higher yields much lower values in this metric than with version 0.9.2. Furthermore, these values do not correspond to the results shown when generating the confusion matrix with torchmetrics.

### To Reproduce

Steps to reproduce the behavior...

Calculate the accuracy metric at the test_step of a classification model. (In our case a densenet variation).

##### Test step with torchmetrics 0.9.2:

```
def test_step(self, test_batch, batch_idx):
        outputs, labels = test_batch
        results = self(outputs)
        loss = F.cross_entropy(results, labels)
        accuracy = torchmetrics.functional.accuracy(results, labels)

        self.log("test_loss", loss, on_epoch=True)
        self.log("test_accuracy", accuracy, on_epoch=True)

        return {'loss': loss, 'accuracy': accuracy, 'results': results.detach(), 'labels': labels.detach()}
```
Output:
```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃            Test metric             ┃            DataLoader 0            ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│              F1_macro              │         0.8626117706298828         │
│           test_accuracy            │         0.9467542171478271         │
│     test_accuracy_left_attack      │         0.9508514404296875         │
│ test_accuracy_left_counter_attack  │         0.9349775910377502         │
│     test_accuracy_left_penalty     │         0.9522821307182312         │
│     test_accuracy_right_attack     │         0.9421581625938416         │
│ test_accuracy_right_counter_attack │         0.8808988928794861         │
│    test_accuracy_right_penalty     │         0.8948717713356018         │
│       test_accuracy_timeout        │         0.988043487071991          │
│             test_loss              │        0.18138954043388367         │
└────────────────────────────────────┴────────────────────────────────────┘
```

##### Test step with torchmetrics 0.10.0 and later:

```
def test_step(self, test_batch, batch_idx):
        outputs, labels = test_batch
        results = self(outputs)
        loss = F.cross_entropy(results, labels)
        accuracy = torchmetrics.functional.classification.multiclass_accuracy(results, labels, num_classes=len(self.classes))

        self.log("test_loss", loss, on_epoch=True)
        self.log("test_accuracy", accuracy, on_epoch=True)

        return {'loss': loss, 'accuracy': accuracy, 'results': results.detach(), 'labels': labels.detach()}
```
Output:
```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃            Test metric             ┃            DataLoader 0            ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│              F1_macro              │         0.8225042819976807         │
│           test_accuracy            │        0.13341589272022247         │
│     test_accuracy_left_attack      │         0.956867516040802          │
│ test_accuracy_left_counter_attack  │         0.7825112342834473         │
│     test_accuracy_left_penalty     │         0.5679460763931274         │
│     test_accuracy_right_attack     │         0.9668290615081787         │
│ test_accuracy_right_counter_attack │         0.8426966071128845         │
│    test_accuracy_right_penalty     │         0.8817307949066162         │
│       test_accuracy_timeout        │         0.9889130592346191         │
│             test_loss              │         0.2517324984073639         │
└────────────────────────────────────┴────────────────────────────────────┘
```

### Expected behavior

The obtained accuracy is the same and have sense in both versions and corresponds with the other metrics.

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source): 0.9.2 and 0.10.0 (pip)
- Python & PyTorch Version (e.g., 1.0): 1.13.1+cu117
- Any other relevant information such as OS (e.g., Linux): Linux

**Comment by SkafteNicki** [comment]
Hi @Kr4is,

Sorry for the delay in response from my side. I assume this is because the default value of `average` was changed from `"micro"` to `"macro"`. Therefore, please replace
```python
accuracy = torchmetrics.functional.accuracy(results, labels)
```python
in your code with
accuracy = torchmetrics.functional.classification.multiclass_accuracy(
  results, labels, num_classes=len(self.classes), average="micro"
)
```
which will give you the same result.
